src/ape/utils.py

```diff
@@ -513,45 +557,106 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     Returns:
         List[pathlib.Path]: A list of files in the given directory.
     """
+
     if path.is_dir():
         return list(path.rglob("*.*"))
 
     return [path]
 
 
-class AbstractDataClassMeta(DataClassMeta, ABCMeta):
+class injected_before_use(property):
+    """
+    Injected properties are injected class variables that must be set before use
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
+    def __get__(self, *args):
+        raise ValueError("Value not set. Please inject this property before calling.")
+
+
+class ManagerAccessBase:
```

Comment:
Missing docs for this ?

Is this public? Do we expected users to imports / use this, perhaps when making plugins?

Otherwise, we should consider renaming it to `_ManagerAccessBase` and intentionally hiding it from the docs...

Reply:
Explore how docs are structured in another PR